core/src/apps/management/reboot_to_bootloader.py

```diff
@@ -7,20 +7,65 @@
 
 
 async def reboot_to_bootloader(msg: RebootToBootloader) -> NoReturn:
-    from trezor import io, loop, utils
+    from trezor import io, loop, utils, wire
+    from trezor.enums import BootCommand
     from trezor.messages import Success
     from trezor.ui.layouts import confirm_action
     from trezor.wire.context import get_context
 
-    await confirm_action(
-        "reboot",
-        "Go to bootloader",
-        "Do you want to restart Trezor in bootloader mode?",
-        verb="Restart",
-    )
-    ctx = get_context()
-    await ctx.write(Success(message="Rebooting"))
-    # make sure the outgoing USB buffer is flushed
-    await loop.wait(ctx.iface.iface_num() | io.POLL_WRITE)
-    utils.reboot_to_bootloader()
-    raise RuntimeError
+    if msg.boot_command is None or msg.boot_command == BootCommand.STOP_AND_WAIT:
+        await confirm_action(
+            "reboot",
+            "Go to bootloader",
+            "Do you want to restart Trezor in bootloader mode?",
+            verb="Restart",
+        )
+        ctx = get_context()
+        await ctx.write(Success(message="Rebooting"))
+        # make sure the outgoing USB buffer is flushed
+        await loop.wait(ctx.iface.iface_num() | io.POLL_WRITE)
+        utils.reboot_to_bootloader(BootCommand.STOP_AND_WAIT)
+        raise RuntimeError
+
+    elif (
+        msg.boot_command == BootCommand.INSTALL_UPGRADE
+        and msg.firmware_header is not None
+    ):
```

Comment:
for slightly better forward compatibility, let's make this the main `if` branch, and the other version an `else` branch (so that an unrecognized argument goes to bootloader the usual way)

Reply:
refactored by @mmilata commit, see f62afd3947755a0645b92be72ca3208b175cdf2e